services/121-service/src/registration/registrations.service.ts

```diff
@@ -760,24 +762,77 @@ export class RegistrationsService {
       );
 
       if (message) {
-        this.sendSmsMessage(registration, programId, message);
+        this.sendTextMessage(registration, programId, message);
       }
     }
   }
 
-  private async sendSmsMessage(
+  private async sendTextMessage(
     registration: RegistrationEntity,
     programId: number,
     message?: string,
+    key?: string,
   ): Promise<void> {
-    this.smsService.notifyBySms(
-      registration.id,
-      registration.phoneNumber,
-      registration.preferredLanguage,
-      programId,
-      message,
-      null,
-    );
+    if (!message && !key) {
+      throw new HttpException(
+        'A message or a key should be supplied.',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    let whatsappNumber;
+    if (registration.customData) {
+      whatsappNumber =
+        registration.customData[CustomDataAttributes.whatsappPhoneNumber];
+    }
+    if (!registration.phoneNumber && !whatsappNumber) {
+      throw new HttpException(
```

Comment:
there is no proper error handling on this in the frontend right?

Reply:
No, I guess not. I copied it from here. https://github.com/global-121/121-platform/pull/2610/files#diff-5b540315f0142d5a1139836201d057c7efc35f46c21defdcf45bbe93f95504ceL22

Reply:
So it was already in the code. I should never be triggered. Maybe we can leave it in so it's nice for debugging